engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageState.java

```diff
@@ -274,6 +322,11 @@ public void remove(final long key) {
     deadline.wrapLong(storedMessage.getMessage().getDeadline());
     deadlineColumnFamily.deleteExisting(deadlineMessageKey);
 
+    localMessageDeadlineCount -= 1L;
+    messagesDeadlineCount.wrapLong(localMessageDeadlineCount);
+    messagesDeadlineCountColumnFamily.update(messagesDeadlineCountKey, messagesDeadlineCount);
```

Comment:
:x: As far as I can tell this `update` is not correct. When we introduce this now, we will have messages that are not yet counted. When one of those is removed before a new one is inserted, this `update` will throw!